Cubical/Structures/Group/Algebra.agda

```diff
@@ -0,0 +1,229 @@
+  isInj-f x y p = sym (leftInv f _) ∙∙ cong g p ∙∙ leftInv f _
+
+invGroupIso : {G : Group {ℓ}} {H : Group {ℓ'}} → GroupIso G H → GroupIso H G
+fun (map (invGroupIso iso1)) = inv iso1
+isHom (map (invGroupIso iso1)) = isGroupHomInv' iso1
+inv (invGroupIso iso1) = fun (map iso1)
+rightInv (invGroupIso iso1) = leftInv iso1
+leftInv (invGroupIso iso1) = rightInv iso1
+
+_◆_ : _
+_◆_ = compGroupIso
+
+dirProdGroupIso : ∀ {ℓ'' ℓ'''} {G : Group {ℓ}} {H : Group {ℓ'}} {A : Group {ℓ''}} {B : Group {ℓ'''}}
+               → GroupIso G H → GroupIso A B → GroupIso (dirProd G A) (dirProd H B)
+fun (map (dirProdGroupIso iso1 iso2)) prod = fun (map iso1) (fst prod) , fun (map iso2) (snd prod)
+isHom (map (dirProdGroupIso iso1 iso2)) a b = ΣPathP (isHom (map iso1) (fst a) (fst b) , isHom (map iso2) (snd a) (snd b))
+inv (dirProdGroupIso iso1 iso2) prod = (inv iso1) (fst prod) , (inv iso2) (snd prod)
+rightInv (dirProdGroupIso iso1 iso2) a = ΣPathP (rightInv iso1 (fst a) , (rightInv iso2 (snd a)))
+leftInv (dirProdGroupIso iso1 iso2) a = ΣPathP (leftInv iso1 (fst a) , (leftInv iso2 (snd a)))
+
+GrIsoToGrEquiv : {G : Group {ℓ}} {H : Group {ℓ'}} → GroupIso G H → GroupEquiv G H
+GroupEquiv.eq (GrIsoToGrEquiv i) = isoToEquiv (iso (fun (map i)) (inv i) (rightInv i) (leftInv i))
+GroupEquiv.isHom (GrIsoToGrEquiv i) = isHom (map i)
+
+--- Proofs that BijectionIso and vSES both induce isomorphisms ---
+BijectionIsoToGroupIso : {A : Group {ℓ}} {B : Group {ℓ'}} → BijectionIso A B → GroupIso A B
+BijectionIsoToGroupIso {A = A} {B = B} i = grIso
+  where
+  module A = Group A
+  module B = Group B
+  f = fun (map' i)
+
+  helper : (b : _) → isProp (Σ[ a ∈ ⟨ A ⟩ ] f a ≡ b)
+  helper _ a b =
+    Σ≡Prop (λ _ → isSetCarrier B _ _)
+           (fst a ≡⟨ sym (IsGroup.rid (isGroup A) (fst a)) ⟩
+           ((fst a) A.+ 0g A) ≡⟨ cong ((fst a) A.+_) (sym (invl A (fst b))) ⟩
+           ((fst a) A.+ ((A.- fst b) A.+ fst b)) ≡⟨ Group.assoc A _ _ _ ⟩
+           (((fst a) A.+ (A.- fst b)) A.+ fst b) ≡⟨ cong (A._+ fst b) idHelper ⟩
+           (0g A A.+ fst b) ≡⟨ IsGroup.lid (isGroup A) (fst b) ⟩
+           fst b ∎)
+    where
+    idHelper : fst a A.+ (A.- fst b) ≡ 0g A
+    idHelper =
+      inj i _
+           (isHom (map' i) (fst a) (A.- (fst b))
+         ∙ (cong (f (fst a) B.+_) (morphMinus A B (map' i) (fst b))
+         ∙∙ cong (B._+ (B.- f (fst b))) (snd a ∙ sym (snd b))
+         ∙∙ invr B (f (fst b))))
+
+  grIso : GroupIso A B
+  map grIso = map' i
+  inv grIso b = (rec (helper b) (λ a → a) (surj i b)) .fst
+  rightInv grIso b = (rec (helper b) (λ a → a) (surj i b)) .snd
+  leftInv grIso b j = rec (helper (f b)) (λ a → a) (propTruncIsProp (surj i (f b)) ∣ b , refl ∣ j) .fst
+
+BijectionIsoToGroupEquiv : {A : Group {ℓ}} {B : Group {ℓ'}} → BijectionIso A B → GroupEquiv A B
+BijectionIsoToGroupEquiv i = GrIsoToGrEquiv (BijectionIsoToGroupIso i)
+
+vSES→GroupIso : ∀ {ℓ ℓ' ℓ'' ℓ'''} {A : Group {ℓ}} {B : Group {ℓ'}} (leftGr : Group {ℓ''}) (rightGr : Group {ℓ'''})
```

Comment:
Can't you quantify over levels at the top? Also, \ell''' isn't very nice... How about using \_3 instead?